EKF/estimator_interface.cpp

```diff
@@ -391,21 +391,56 @@ void EstimatorInterface::setOpticalFlowData(uint64_t time_usec, flow_message *fl
 
 			// NOTE: the EKF uses the reverse sign convention to the flow sensor. EKF assumes positive LOS rate is produced by a RH rotation of the image about the sensor axis.
 			// copy the optical and gyro measured delta angles
-			optflow_sample_new.gyroXYZ = - flow->gyrodata;
+
+			bool no_gyro = false;
+			imuSample matching_imu_sample;
+			Vector3f matching_gyro_sample;
+
+			/* for optical flow boards with no gyro. The driver will set the gyro x, y and z fields in the optical_flow message to NAN so that  
+			 * we can check this here and get the gyro data with the matching timestamp from the imu buffer. The gyro data is divided by dt to 
+			 * convert it into a rate before removing it from the flow to remove the rotation component. This is done to prevent errors when the flow
+			 * and gyro data are integrated over different times */
+			if (!ISFINITE(flow->gyrodata(0)) && !ISFINITE(flow->gyrodata(1)) && !ISFINITE(flow->gyrodata(2))) {
```

Comment:
Would it be worth doing OR here to catch any bogus output in the regular optical flow case?

Reply:
@dagar can you please explain the specific error condition you would like to see addressed here.

Reply:
The current logic switches to using gyro from the EKF if all 3 flow->gyrodata fields are not finite. I was asking about switching it to ANY flow->gyrodata fields being non-finite, which would prevent bogus data from entering in regular usage without additional checks.

Reply:
OK, that makes sense